File: ruby/lib/casilla_calle.rb
module Civitas
    class CasillaCalle < Casilla
        def initialize(titulo)
            super(titulo.nombre)
            @importe=titulo.precioCompra
            @tituloPropiedad=titulo
        end
        attr_reader :importe, :tituloPropiedad

        def recibeJugador(actual, todos)
            if (jugadorCorrecto(actual, todos))
                informe(actual, todos)
                jugador = todos[actual]
                if !@tituloPropiedad.tienePropietario()
                    jugador.puedeComprarCasilla
                else
                    @tituloPropiedad.tramitarAlquiler(jugador)
                end
            end
        end
    end
end

File: Civitas_gui/src/civitas/CasillaCalle.java
package civitas;
import java.util.ArrayList;

public class CasillaCalle extends Casilla {
    private TituloPropiedad tituloPropiedad;
    private float importe;

    CasillaCalle(TituloPropiedad titulo) {
        super(titulo.getNombre());
        tituloPropiedad = titulo;
        importe = titulo.getPrecioCompra();
    }
    
    @Override
    public float getImporte(){
        return importe;
    }

    public TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    
    @Override
    public String getTipo(){
        return "Calle";
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            Jugador jugador = todos.get(actual);
            if (!tituloPropiedad.tienePropietario()) {
                jugador.puedeComprarCasilla();
            } else {
                tituloPropiedad.tramitarAlquiler(jugador);
            }
        }
    }
}